boltdesignsystem/bolt
# DS 835: Table Element

## Jira

https://pegadigitalit.atlassian.net/browse/DS-835

## Summary

The new element `table` is created that can replace the basic functionalities of the current Table Component

## Details
1. Much of the props and functionality are taken from the old table component, for example:

- only rows table
- table with top headers only
- table with side headers only
- table with both top and side headers
- caption of the table
- borderless table
- format table (regular, numeric)

2. The following methods of display

- Twig (appearance and behavior can be achieved via choosing props from the schema)
- HTML only (appearance and behavior can be achieved by adding appropriate classes to the `<table>`)
- HTML only (without the **tbody**) (appearance and behavior can be achieved by adding appropriate classes to the `<table>`)

3. JavaScript creates a wrapper and wraps the `<table>` if needed
4. There are props that set the first, second, or first two columns as fixed width
5. There is now a full width prop that will changed the default behavior (inline-block) to be full width
6. There are sticky header functionality now included

## How to test

(Focus on testing in all browsers)
* Pull the branch.
* Check the code, especially the CSS and JS. 
* Check the `table-element` docs.
* Go to test folder and find table test examples. 
* Play with them and confirm they work properly. 

## Release notes

There is a new Table Element that replaces the basic functionalities of the current Table Component
